### PR TITLE
Changes for IP-Intelligence and Firewall Configuration

### PR DESCRIPTION
**Pull request by twainpereira**

### CODE REVIEW

File: f5/bigip/tm/security/ip_intelligence.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+class Blacklist_Category(Resource):
+    """BIG-IP® AFM® IP-Intelligence Blacklist Category resource"""
+
+    def __init__(self, blacklist_categorys):
+        super(Blacklist_Category, self).__init__(blacklist_categorys)
+        self._meta_data['required_json_kind'] = \
+            'tm:security:ip-intelligence:blacklist-category:blacklist-categorystate'
+        self._meta_data['required_creation_parameters'].update(
+            ('partition', 'name'))
```
Comment (wojtek0806) [inline f5/bigip/tm/security/ip_intelligence.py]: 'name' not required its inherited in resource, please remove

File: f5/bigip/tm/security/ip_intelligence.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        self._meta_data['required_load_parameters'].update(
+            ('partition', 'name'))
```
Comment (wojtek0806) [inline f5/bigip/tm/security/ip_intelligence.py]: name not required please remove

File: f5/bigip/tm/security/firewall.py [hunk 3]
```diff
@@ -232,3 +233,11 @@ def __init__(self, global_rules):
         super(Global_Rules, self).__init__(global_rules)
         self._meta_data['required_json_kind'] = \
             'tm:security:firewall:global-rules:global-rulesstate'
+
+
+class Global_Fqdn_Policy(UnnamedResource):
+    """BIG-IP® AFM® Global FQDN policy resource"""
+    def __init__(self, global_fqdn_policy):
```
Comment (wojtek0806) [inline f5/bigip/tm/security/firewall.py:240]: if this collection works really on 12.x then minimal version needs to be added here, by default we have it set on 11.5.0:

https://github.com/F5Networks/f5-common-python/blob/development/f5/bigip/resource.py#L169

You set it on collections (not Organizing Collections) or unnamed resources:

https://github.com/F5Networks/f5-common-python/blob/development/f5/bigip/tm/asm/policies/vulnerabilities.py#L27

File: f5/bigip/tm/security/ip_intelligence.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+
+class Policy(Resource):
+    """BIG-IP® AFM® IP-Intelligence Policy resource"""
+
+    def __init__(self, policy_s):
+        super(Policy, self).__init__(policy_s)
+        self._meta_data['required_json_kind'] = \
+            'tm:security:ip-intelligence:policy:policystate'
+        self._meta_data['required_creation_parameters'].update(('partition', 'name'))
```
Comment (wojtek0806) [inline f5/bigip/tm/security/ip_intelligence.py]: remove name

File: f5/bigip/tm/security/ip_intelligence.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        self._meta_data['required_load_parameters'].update(('partition', 'name'))
```
Comment (wojtek0806) [inline f5/bigip/tm/security/ip_intelligence.py]: remove name

**Review by wojtek0806 — CHANGES_REQUESTED** [review]
After you make the corrections please squash commits

File: f5/bigip/tm/security/firewall.py (hunk 3 above)
Comment (twainpereira) [reply f5/bigip/tm/security/firewall.py:240]: Added in version check for Global FQDN policy

File: f5/bigip/tm/security/ip_intelligence.py [hunk 6]
```diff
@@ -110,10 +106,6 @@ def __init__(self, blacklist_categorys):
         super(Blacklist_Category, self).__init__(blacklist_categorys)
         self._meta_data['required_json_kind'] = \
             'tm:security:ip-intelligence:blacklist-category:blacklist-categorystate'
-        self._meta_data['required_creation_parameters'].update(
```
Comment (wojtek0806) [inline f5/bigip/tm/security/ip_intelligence.py]: is the partition parameter mandatory for this if so only name should have been removed and partition should be added

File: f5/bigip/tm/security/ip_intelligence.py [hunk 7]
```diff
@@ -110,10 +106,6 @@ def __init__(self, blacklist_categorys):
...
-            ('partition', 'name'))
-        self._meta_data['required_load_parameters'].update(
```
Comment (wojtek0806) [inline f5/bigip/tm/security/ip_intelligence.py]: is the partition parameter mandatory for this if so only name should have been removed and partition should be added

File: f5/bigip/tm/security/ip_intelligence.py [hunk 8]
```diff
@@ -62,8 +62,6 @@ def __init__(self, feed_list_s):
         super(Feed_list, self).__init__(feed_list_s)
         self._meta_data['required_json_kind'] = \
             'tm:security:ip-intelligence:feed-list:feed-liststate'
-        self._meta_data['required_creation_parameters'].update(('partition',))
```
Comment (wojtek0806) [inline f5/bigip/tm/security/ip_intelligence.py]: is the partition parameter mandatory for this if so only name should have been removed and partition should be added not removed

File: f5/bigip/tm/security/ip_intelligence.py [hunk 9]
```diff
@@ -62,8 +62,6 @@ def __init__(self, feed_list_s):
...
-        self._meta_data['required_load_parameters'].update(('partition',))
```
Comment (wojtek0806) [inline f5/bigip/tm/security/ip_intelligence.py]: is the partition parameter mandatory for this if so only name should have been removed and partition should be added

**Comment by wojtek0806** [comment]
please squash commits

**Review by wojtek0806 — APPROVED** [review]
Looks good to me

**Pull request merged**